youtube_dl/extractor/nhl.py

```diff
@@ -349,3 +354,201 @@ def _real_extract(self, url):
+            'Downloading game data',
+            'Unable to download game data')
+
+    def get_stream_data(self, video_id):
+        stream_data_url = 'https://mf.svc.nhl.com/ws/media/mf/v2.4/stream?contentId=%s&playbackScenario=HTTP_CLOUD_WIRED_WEB&sessionKey=%s&auth=response&format=json&platform=WEB_MEDIAPLAYER&_=%s000'
+        session_key = 'abcdefghijklmnop'
+        timestamp = int(time.time())
+        url = stream_data_url % (video_id, session_key, timestamp)
+        return self.extractor._download_json(url, video_id, 'Downloading stream data', 'Unable to download stream data')
+
+
+class NHLTVIE(InfoExtractor):
+    IE_NAME = 'nhl.com:nhltv'
+    _VALID_URL = r'https?://(?:www\.)?nhl.com/tv/(?P<gameId>\d+)(/[^/]+)*(/(?P<id>\d+))?'
+    _NETRC_MACHINE = 'nhltv'
+    _NHLTV_CACHE = 'nhltv'
+    _TESTS = [{
+        # This is a free video that anyone with an NHL TV login can access
+        'url': 'https://www.nhl.com/tv/2016020321/221-1003765/46561403',
+        'md5': '34d9518c495ebdad947b9723b5a7c9a9',
+        'info_dict': {
+            'id': '46561403',
+            'ext': 'ts',
+            'title': '2016-11-26: Anaheim Ducks @ San Jose Sharks (AWAY feed)',
+            'timestamp': 1480217427,
+        },
+        'skip': 'requires a login to NHL TV, and takes a long time to run as the video is 3h+',
+        'params': {
+            'usenetrc': True,
+            'format': '[width=400]',
+            'hls_use_mpegts': True,
+        }
+    }, {
+        'url': 'https://www.nhl.com/tv/2016020362/221-1003808',
+        'only_matching': True,
+    }, {
+        'url': 'https://www.nhl.com/tv/2016020373',
+        'only_matching': True,
+    }]
+
+    def _real_initialize(self):
+        auth = self._downloader.cache.load(self._NHLTV_CACHE, 'auth')
+        self.api = NHLApi(self, auth)
+        if auth is None:
+            self._login()
+
+    def _login(self):
+        (username, password) = self._get_login_info()
+        if username is None:
+            self.raise_login_required()
+        auth_provider = self._get_auth_provider() or 'nhl'
+        self.report_login()
+        self.api.login(username, password, auth_provider)
+        self._downloader.cache.store(self._NHLTV_CACHE, 'auth', self.api.auth)
+
+    def _real_extract(self, url):
+        mobj = re.match(self._VALID_URL, url)
+        video_id, game_id = mobj.group('id'), mobj.group('gameId')
+        # At this point we may have a game_id without a video_id. The next call
+        # ensures that we get a video_id for the desired game.
+        video_id, title, timestamp = self.extract_game_info(video_id, game_id)
+        m3u8_url, media_auth = self.extract_playlist_url_and_auth(video_id)
+        m3u8_url = self.get_60fps_playlist(m3u8_url)
+
+        # media auth cookie authenticates the specific download url
+        self._set_cookie('nhl.com', 'mediaAuth_v2', media_auth)
+        formats = self._extract_m3u8_formats(m3u8_url, video_id, 'ts', m3u8_id='hls')
```

Comment:
I disagree. The file created is not an mp4 container, it's a copy of an mpeg2ts stream. They are different formats. `.ts` is recognized by mpv and vlc and likely other media players as a stream format that may not yet be completely finished. I've had problems using the `.mp4` in the past with media players that expect headers to give information on stream length. This avoids that problem.